crud/common/utils.lua

```diff
@@ -105,19 +106,44 @@ function utils.get_space(space_name, replicasets)
             'perhaps other instances are unavailable or you have configured only the router')
     end
 
-    if replicaset.master == nil then
-        local error_msg = string.format(
-            'The master was not found in replicaset %s, ' ..
-            'check status of the master and repeat the operation later',
-             replicaset.uuid)
-        return nil, GetSpaceError:new(error_msg)
-    end
+    if replicaset.master == nil or replicaset.master.conn.error ~= nil then
+        -- Try to get master again with timeout.
+        local deadline = fiber_clock() + const.GET_SPACE_TIMEOUT
+        local vshard_router, err = utils.get_vshard_router_instance()
+        if err ~= nil then
+            return nil, GetSpaceError:new(err)
+        end
+        while true do
+            ::continue::
+            fiber.yield()
+            replicasets = vshard_router:routeall()
+            replicaset = select(2, next(replicasets))
+
+            if replicaset.master ~= nil and replicaset.master.conn.error ~= nil then
```

Comment:
Reducing code copypaste also is a way to reduce the possibility of bugs. For example, here you've forgot to checks that replicasets aren't nil even though this check is already implemented.

Reply:
Done, now `replicaset ~= nil` check there is inside timeout loop